src/app/providers.tsx

```diff
@@ -28,6 +34,31 @@ createWeb3Modal({
 })
 
 export function Providers({ children }: { children: React.ReactNode }) {
+  const [persistedUserData, setPersistedUserData] = useState<User | null>(null)
+
+  useEffect(() => {
+    const unwatch = watchAccount(config, {
+      async onChange(account) {
+        try {
+          const user = await (
+            await fetch('/api/user', {
+              method: 'POST',
```

Comment:
could this be a `GET`?

Reply:
well we could define both. I don't think it matters much, though here' we are always updating the user's connect timestamp, and also we create the user if it doesn't exist already